genesis/engine/solvers/rigid/gjk_decomp.py

```diff
@@ -0,0 +1,3394 @@
+
+                    break
+
+            # Only try shrinking the sphere once
+            shrink_sphere = False
+
+        distance = self.distance[i_b]
+        nsimplex = self.gjk_nsimplex[i_b]
+        collided = distance < self.collision_eps
+
+        # To run EPA, we need following conditions:
+        # 1. We did not find min. distance with shrink_sphere flag
+        # 2. We have a valid GJK simplex (nsimplex > 0)
+        # 3. We have a collision (distance < collision_epsilon)
+        do_epa = (not shrink_sphere) and collided and (nsimplex > 0)
+
+        if do_epa:
+
+            # Assume touching
+            self.distance[i_b] = 0
+
+            # Initialize polytope
+            self.polytope[i_b].nverts = 0
+            self.polytope[i_b].nfaces = 0
+            self.polytope[i_b].nfaces_map = 0
+            self.polytope[i_b].horizon_nedges = 0
+
+            # Construct the initial polytope from the GJK simplex
+            polytope_flag = 0
+            if nsimplex == 2:
+                polytope_flag = self.func_epa_init_polytope_2d(i_ga, i_gb, i_b)
+            elif nsimplex == 4:
+                polytope_flag = self.func_epa_init_polytope_4d(i_ga, i_gb, i_b)
+
+            # Polytope 3D could be used as a fallback for 2D and 4D cases, but it is not necessary
+            if nsimplex == 3 or (polytope_flag == EPA_P2_FALLBACK3 or polytope_flag == EPA_P4_FALLBACK3):
+                polytope_flag = self.func_epa_init_polytope_3d(i_ga, i_gb, i_b)
+
+            # Run EPA from the polytope
+            if polytope_flag == 0:
+                i_f = self.func_epa(i_ga, i_gb, i_b)
+                distance = self.distance[i_b]
+
+                if num_max_contacts > 1 and i_f != -1:
+                    # Detect multiple contacts
+                    self.func_multi_contact(i_ga, i_gb, i_b, i_f)
+
+    """
+    GJK algorithms
+    """
+
+    @ti.func
+    def func_gjk(self, i_ga, i_gb, i_b, shrink_sphere):
+        """
+        GJK algorithm to compute the minimum distance between two convex objects.
+        """
+
+        # Simplex index
+        n = 0
+
+        # Final number of simplex vertices
+        nsimplex = 0
+
+        # Number of witness points and distance
+        nx = 0
+        dist = 0.0
+
+        # Lambda for barycentric coordinates
+        _lambda = gs.ti_vec4(1.0, 0.0, 0.0, 0.0)
+
+        # Final return flag
+        return_flag = 0
+
+        # Set initial guess of support vector using the positions.
+        # The support vector should be a non-zero vector.
+        approx_witness_point_obj1 = self._solver.geoms_state[i_ga, i_b].pos
+        approx_witness_point_obj2 = self._solver.geoms_state[i_gb, i_b].pos
+        support_vector = approx_witness_point_obj1 - approx_witness_point_obj2
+
+        if support_vector.dot(support_vector) < self.FLOAT_MIN_SQ:
+            approx_witness_point_obj1.x = approx_witness_point_obj1.x + 0.01
```

Comment:
At this point, why not just write `support_vector = ti.Vector([1.0, 0.0, 0.0], dt=gs.ti_float)` directly?